File: whdon_app/src/main/java/nl/whitedove/washetdroogofniet/StatsPerPlaatsActivity.java
package nl.whitedove.washetdroogofniet;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.ListView;

import java.lang.ref.WeakReference;
import java.util.ArrayList;

public class StatsPerPlaatsActivity extends Activity {

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.per_plaats_statistieken);
        InitFab();
        ToondataBackground();
    }

    private void InitFab() {
        FloatingActionButton fabTerug = findViewById(R.id.btnTerug);
        fabTerug.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Terug();
            }
        });
    }

    private void Stat1Plaats(String locatie) {
        Intent intent = new Intent(this, Stats1PlaatsActivity.class);
        intent.putExtra("Locatie", locatie);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
    }

    private void Terug() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
    }

    private void ToondataBackground() {
        Context context = getApplicationContext();
        new AsyncGetStatistiekenPerPlaatsTask(this).execute(context);
    }

    private void ToonPerPlaatsStatistieken(ArrayList<Statistiek> stats) {
        if (stats == null || stats.size() <= 1) {
            return;
        }
        final ListView lvStats = findViewById(R.id.lvStats);

        final CustomListAdapterTotStats adapter = new CustomListAdapterTotStats(this, stats);
        lvStats.setAdapter(adapter);

        EditText etZoek = findViewById(R.id.etZoek);
        etZoek.addTextChangedListener(new TextWatcher() {

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                adapter.getFilter().filter(s.toString());
            }

            @Override
            public void beforeTextChanged(CharSequence s, int start, int count,
                                          int after) {
            }

            @Override
            public void afterTextChanged(Editable s) {
            }
        });

        lvStats.setOnItemClickListener(new AdapterView.OnItemClickListener() {

            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                Statistiek stat = (Statistiek) parent.getItemAtPosition(position);
                Stat1Plaats(stat.getLocatie());
            }
        });

    }

    private static class AsyncGetStatistiekenPerPlaatsTask extends AsyncTask<Context, Void, ArrayList<Statistiek>> {
        private WeakReference<StatsPerPlaatsActivity> activityWeakReference;

        AsyncGetStatistiekenPerPlaatsTask(StatsPerPlaatsActivity context) {
            activityWeakReference = new WeakReference<>(context);
        }

        @Override
        protected ArrayList<Statistiek> doInBackground(Context... params) {
            Context context = params[0];
            DatabaseHelper dh = DatabaseHelper.Companion.getInstance(context);
            return dh.GetStatistieken();
        }

        @Override
        protected void onPostExecute(ArrayList<Statistiek> stats) {

            StatsPerPlaatsActivity activity = activityWeakReference.get();
            if (activity != null) activity.ToonPerPlaatsStatistieken(stats);
        }

    }
}

File: whdon_app/src/main/java/nl/whitedove/washetdroogofniet/StatsPerPlaatsActivity.kt
package nl.whitedove.washetdroogofniet

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.text.Editable
import android.text.TextWatcher
import android.widget.AdapterView
import android.widget.EditText
import android.widget.ListView

import java.lang.ref.WeakReference
import java.util.ArrayList

class StatsPerPlaatsActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.per_plaats_statistieken)
        initFab()
        toondataBackground()
    }

    private fun initFab() {
        val fabTerug = findViewById<FloatingActionButton>(R.id.btnTerug)
        fabTerug.setOnClickListener { terug() }
    }

    private fun stat1Plaats(locatie: String?) {
        val intent = Intent(this, Stats1PlaatsActivity::class.java)
        intent.putExtra("Locatie", locatie)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
        startActivity(intent)
    }

    private fun terug() {
        val intent = Intent(this, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
        startActivity(intent)
    }

    private fun toondataBackground() {
        val context = applicationContext
        AsyncGetStatistiekenPerPlaatsTask(this).execute(context)
    }

    private fun toonPerPlaatsStatistieken(stats: ArrayList<Statistiek>?) {
        if (stats == null || stats.size <= 1) {
            return
        }
        val lvStats = findViewById<ListView>(R.id.lvStats)

        val adapter = CustomListAdapterTotStats(this, stats)
        lvStats.adapter = adapter

        val etZoek = findViewById<EditText>(R.id.etZoek)
        etZoek.addTextChangedListener(object : TextWatcher {

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                adapter.filter.filter(s.toString())
            }

            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int,
                                           after: Int) {
            }

            override fun afterTextChanged(s: Editable) {}
        })

        lvStats.onItemClickListener = AdapterView.OnItemClickListener { parent, _, position, _ ->
            val stat = parent.getItemAtPosition(position) as Statistiek
            stat1Plaats(stat.locatie)
        }

    }

    private class AsyncGetStatistiekenPerPlaatsTask internal constructor(context: StatsPerPlaatsActivity) : AsyncTask<Context, Void, ArrayList<Statistiek>>() {
        private val activityWeakReference: WeakReference<StatsPerPlaatsActivity> = WeakReference(context)

        override fun doInBackground(vararg params: Context): ArrayList<Statistiek> {
            val context = params[0]
            val dh = DatabaseHelper.getInstance(context)
            return dh.GetStatistieken()
        }

        override fun onPostExecute(stats: ArrayList<Statistiek>) {

            val activity = activityWeakReference.get()
            activity?.toonPerPlaatsStatistieken(stats)
        }
    }
}